frontend/src/components/general-comparison-table/general-comparison-table.tsx

```diff
@@ -0,0 +1,128 @@
+import { useLayoutEffect, useState } from 'react';
+
+import { CollapseElement } from '@components/collapse-component/collapse-element/collapse-element';
+import { Spinner } from '@components/common/spinner/spinner';
+import { getElementHeightWithMargins } from '@helpers/utils/get-element-height-with-margins';
+import { useGetComparisonGeneralInfoQuery } from '@store/queries/comparison';
+import { clsx } from 'clsx';
+
+import styles from './styles.module.scss';
+
+const GeneralComparisonTable: React.FC = () => {
+  const { data: generalInfos, isLoading } = useGetComparisonGeneralInfoQuery();
+
+  const options: Set<string> = new Set();
+  generalInfos?.forEach((car) =>
+    Object.keys(car.options).map((optionType: string) =>
+      options.add(optionType),
+    ),
+  );
```

Comment:
This should be in `useMemo` because now `options` are getting re-initialized on each rerender